packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.test.ts

```diff
@@ -0,0 +1,174 @@
+                        const mutation = useShopperBasketsMutation(mutationName)
+
+                        // All of the necessary query hooks needed to verify the cache-update logic
+                        const queries = {
+                            basket: useBasket({basketId: BASKET_ID}),
+                            customerBaskets: useCustomerBaskets({customerId: CUSTOMER_ID})
+                        }
+
+                        return {
+                            queries,
+                            mutation
+                        }
+                    })
+
+                    await waitForValueToChange(() => result.current.queries.basket.data)
+
+                    act(() => {
+                        result.current.mutation.mutate(payload)
+                    })
+
+                    await waitForValueToChange(() => result.current.mutation.isSuccess)
+                    expect(result.current.mutation.isSuccess).toBe(true)
+
+                    // On successful mutation, the query cache gets updated too. Let's assert it.
+                    const cacheUpdateMatrix = getCacheUpdateMatrix(CUSTOMER_ID)
+                    // @ts-ignore
+                    const matrixElement = cacheUpdateMatrix[mutationName](payload, {})
+                    const {invalidate, update, remove}: CacheUpdateMatrixElement = matrixElement
+
+                    update?.forEach(({name}) => {
+                        // @ts-ignore
+                        assertUpdateQuery(result.current.queries[name], {test: 'new data'})
+                    })
+
+                    invalidate?.forEach(({name}) => {
+                        // @ts-ignore
+                        assertInvalidateQuery(result.current.queries[name], {test: 'old data'})
+                    })
+
+                    remove?.forEach(({name}) => {
+                        // @ts-ignore
+                        assertRemoveQuery(result.current.queries[name])
+                    })
+                }
+            },
+            {
+                name: 'error',
+                assertions: async () => {
+                    mockMutationEndpoints('/checkout/shopper-baskets/', {errorResponse: 500})
+
+                    const {result, waitForNextUpdate} = renderHookWithProviders(() => {
+                        return useShopperBasketsMutation(mutationName)
+                    })
+
+                    act(() => {
+                        result.current.mutate(payload)
+                    })
+
+                    await waitForNextUpdate()
+
+                    expect(result.current.error).toBeDefined()
+                }
+            }
+        ]
+    }
+})
+
+tests.forEach(({hook, cases}) => {
+    describe(hook, () => {
+        beforeEach(() => {
+            jest.clearAllMocks()
+        })
+        cases.forEach(({name, assertions}) => {
+            test(name, assertions)
+        })
+    })
+})
+
+const mockRelatedQueries = () => {
```

Comment:
I decided to place this function in this test file and not in the test-utils, because the logic is unique to ShopperBaskets.